core/src/main/java/ch/cyberduck/core/ProtocolFactory.java

```diff
@@ -219,32 +221,91 @@ public Protocol forName(final String identifier, final String provider) {
      * @param provider   Custom inherited protocol definition
      * @return Matching protocol or null if no match
      */
-    public Protocol forName(final List<Protocol> enabled, final String identifier, final String provider) {
-        final Protocol match =
```

Comment:
Removed Streams, because `orElse()` uses a parameter, and is not lazily evaluated.
Thus _all_ streams were evaluated _at once_, before returning.
Changed to range of checks, because `orElseGet` isn't readable.